src/PolyType.TestCases/TestTypes.cs

```diff
@@ -3539,4 +3539,24 @@ public delegate Task<int> LargeAsyncDelegate(
 [GenerateShapeFor<FSharpFunc<Tuple<int, int>, int>>]
 [GenerateShapeFor<FSharpFunc<int, Task<int>>>]
 [GenerateShapeFor<FSharpFunc<int, FSharpFunc<int, FSharpFunc<int, Task<int>>>>>]
-public partial class Witness;
\ No newline at end of file
+public partial class Witness;
+
+// Test case for external private field discovery (issue reproduction)
+// This class is in an external assembly (TestCases) and does NOT have [GenerateShape]
+// but has private fields with [PropertyShape] that should be discovered when referenced
+// from a type with [GenerateShape] in another assembly.
+public class ExternalClassWithPrivateField
```

Comment:
Please move above the `Witness` class, inserting it in a relevant location (e.g. together with other "Class" types in this file).